etcdmain/config.go

```diff
@@ -317,6 +356,141 @@ func (cfg *config) Parse(arguments []string) error {
 		}
 	}
 
+	return nil
+}
+
+func (cfg *config) configFromFile() error {
+	// Read in the contents of the config file.
+	b, err := ioutil.ReadFile(cfg.configFile)
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal the YAML.
+	err = yaml.Unmarshal(b, cfg)
+	if err != nil {
+		return err
+	}
+
+	// Make sure the defaults are set for the missing fields.
+	cfg.setDefaultsIfNeeded()
+
+	// Validate whether conflicting parameters are specified.
+	nSet := 0
```

Comment:
Yes, I guess we can remove the call to setDefaults in configFile case.

Can you please let me know what is your thinking on moving the common validations to Parse function? I mean how do we handle them for both cases (cmd line and config file) as the flags.IsSet won't work for the configFile case.